### PR TITLE
Add v1.4.1 singletons

### PR DESCRIPTION
**Pull request by danielpartida**
L1 singleton 0x41675C099F32341bf84BFc5382aF534df5C7461a L2 singleton 0x29fcB43b46531BcA003ddC8FCB67FFE91900C76

# Thank you for contributing to Spellbook!

Thank you for taking the time to submit code in Spellbook. A few things to consider:
- If you are a first-time contributor, please sign the CLA by copy & pasting **_exactly_** what the bot mentions in PR comment
- Refer to [docs](#spellbook-contribution-docs) section below to answer questions
- Dune team will review submitted PRs as soon as possible

## Spellbook sub-projects

To prepare for Spellbook to scale in size, Dune team has started to pilot sub-projects:
- For a quick overview, see [the GH discussion on the topic](https://github.com/duneanalytics/spellbook/discussions/5238)
    - Please ask questions and continue the conversation there
- The first spell lineage to move: `tokens.erc20` and all upstream spells which help build it
    - **_Note_**: Please navigate to [tokens](https://github.com/duneanalytics/spellbook/tree/main/tokens) root directory to find tokens objects

## Spellbook contribution docs

The [docs](https://github.com/duneanalytics/spellbook/tree/main/docs) directory has been implemented to answer as many questions as possible. Please take the time to reference each `.md` file within this directory to understand how to efficiently contribute & why the repo is designed as it is 🪄

Example questions to be answered:
- What does each property in the [model config block](https://github.com/duneanalytics/spellbook/blob/main/docs/models/model_config_block.md) mean?
- What is the [CI test](https://github.com/duneanalytics/spellbook/blob/main/docs/ci_test/ci_test_overview.md) attached to PRs and how can I best utilize it?
- What are the Spellbook [best practices](https://github.com/duneanalytics/spellbook/blob/main/docs/general/best_practices.md)?

Please navigate through the [docs](https://github.com/duneanalytics/spellbook/tree/main/docs) directory to find as much info as you can.

**Note**: happy to take PRs to improve the docs, let us know 🤝

### CODE REVIEW

**Comment by danielpartida** [comment]
Screenshot 2024-02-20 at 11 14 32

Hi @jeff-dude,

I am updating our Safe spells to have the current Safe version, i.e. 1.4.1. Attached is a screenshot displaying an error I currently have with the PR. I do not see any syntax errors in the implementation. Would you happen to know what the error refers to?

**Comment by danielpartida** [comment]
For visibility @alicecha @peterrliem

**Comment by jeff-dude** [comment]
> Screenshot 2024-02-20 at 11 14 32
> Hi @jeff-dude,
> 
> I am updating our Safe spells to have the current Safe version, i.e. 1.4.1. Attached is a screenshot displaying an error I currently have with the PR. I do not see any syntax errors in the implementation. Would you happen to know what the error refers to?

that error is referring to the cluster running the models has crashed, specifically the nodes assigned to run each model. it's because i was hogging all of the cluster resources to get a large new spell to run in CI here: #5364 

the job is done there now, you should be good to simply rerun your attached workflows 🙏 

sorry about that!

**Comment by danielpartida** [comment]
Thank you for the heads-up @jeff-dude, I reran the script and now the rebuild passed!

File: models/safe/bnb/safe_bnb_safes.sql
Comment (jeff-dude) [inline models/safe/bnb/safe_bnb_safes.sql:1]: @danielpartida please note the `bnb` spell has `prod_exclude` tag, so it's not running in CI or prod at the moment. 

here is PR where it was added. i tried to tag you guys for a heads up. please help look into that bug now and fix here 🙏

**Review by jeff-dude — CHANGES_REQUESTED** [review]

File: models/safe/bnb/safe_bnb_safes.sql
Comment (jeff-dude) [reply models/safe/bnb/safe_bnb_safes.sql:1]: https://github.com/duneanalytics/spellbook/pull/5218

Comment (danielpartida) [reply models/safe/bnb/safe_bnb_safes.sql:1]: Thank you for the heads-up @jeff-dude 

Whenever you have the chance @peterrliem and @alicecha please take a look at #5218 (fixing the bnb spell). This is a blocker for merging our PR for integrating the v1.4.1 Safes

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**